mailbox/store/src/main/java/org/apache/james/mailbox/store/event/EventFactory.java

```diff
@@ -41,7 +41,56 @@
 import org.apache.james.mailbox.model.UpdatedFlags;
 import org.apache.james.mailbox.store.mail.model.Mailbox;
 
+import com.google.common.base.Preconditions;
+
 public class EventFactory {
+    public static class MailboxAddedBuilder {
```

Comment:
what about staged builder ?

Reply:
Staged builder are not good at proposing alrenative build options IMO...

Reply:
almost all your parameter are mandatory, it makes sense to use staged builder in this context.